tests/functional/test_freeze.py

```diff
@@ -321,6 +321,47 @@ def test_freeze_git_clone_srcdir(script, tmpdir):
     ).strip()
     _check_output(result.stdout, expected)
 
+@need_mercurial
+def test_freeze_mrecurial_clone_srcdir(script, tmpdir):
+    """
+    Test freezing a Mercurial clone where setup.py is in a subdirectory
+    relative to the repo root and the source code is in a subdirectory
+    relative to setup.py.
+    """
+    # Returns path to a generated package called "version_pkg"
+    pkg_version = _create_test_package_with_srcdir(script, vcs='hg')
+ 
+    result = script.run(
+        'hg', 'clone', pkg_version, 'pip-test-package',
+        expect_stderr=True,
```

Comment:
I'm unsure, I've copied the convention used by other tests in the file.
If I was to speculate I would guess its because we don't particularly care if external tools output to stderr or not.

Reply:
It looks like the test passes without `expect_stderr=True`, so I would remove it. There are plans to make this more strict in the future (so passing `expect_stderr=True` asserts that something was output to stderr), so it saves some effort for that activity.

Reply:
Have removed all occurrences of  `expect_stderr=True`  inside `test_freeze_mercurial_clone_srcdir()`